docs/docs/theme-ui.md

```diff
@@ -0,0 +1,264 @@
+}
+```
+
+Theme UI uses a `theme` configuration object to provide color, typography, layout, and other shared stylistic values through [React context][].
+This allows components within your site to add styles based on a predefined set of values.
+
+The Theme UI plugin uses the [component shadowing API][] to add the theme object context to your site.
+Create a `src/gatsby-plugin-theme-ui` directory in your project, and add an `index.js` file to export a theme object.
+
+```shell
+mkdir src/gatsby-plugin-theme-ui
+```
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {}
+```
+
+## Creating a theme object
+
+Add a `colors` object to the file created above to store the color palette for your site.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+}
+```
+
+Next add some base typographic values.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  // highlight-start
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-end
+}
+```
+
+Next add values for use in margin and padding.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-start
+  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
+  // highlight-end
+}
+```
+
+Feel free to include as many additional values as you'd like in your theme.
+Read more about theming in the [Theme UI documentation](https://theme-ui.com/theming).
+
+## Adding styles to elements
+
+Theme UI uses a custom [JSX pragma][] comment to add support for Emotion's `css` prop,
+which can be used to style elements.
+By default JSX compiles to the `React.createElement` function.
+When you add the `/** @jsx jsx */` comment to the top of a file, JSX will compile to the function name in the comment instead.
+For example, `<button />` would compile to `jsx('button')`.
+Theme UI's `jsx` function still renders React elements but lets you use the `css` prop for styling.
+
+[jsx pragma]: https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#pragma
+
+The following example uses the `css` prop to add styles to an element.
```

Comment:
Listing these two code examples first in the Adding Styles to Elements section and then explaining the advanced pragma stuff could strike a nice balance between providing information and not overwhelming people who just want to read code and try stuff out. Then, we could add an `### Advanced tips` heading and the rest of the explanation. How does that sound to you?